assembly/src/assembler/mod.rs

```diff
@@ -202,6 +204,15 @@ impl Assembler {
     }
 
     /// Adds the compiled library to provide modules for the compilation.
+    ///
+    /// We only current support adding non-vendored libraries - that is, the source code of exported
+    /// procedures is not included in the program that compiles against the library. The library's
+    /// source code is instead expected to be loaded in the processor at execution time. Hence, all
+    /// calls to library procedures will be compiled down to a [`vm_core::mast::ExternalNode`] (i.e.
+    /// a reference to the procedure's MAST root). This means that when executing a program compiled
+    /// against a library, the processor will not be able to differentiate procedures with the same
+    /// MAST root but different decorators. Hence, it is not recommended to export two procedures
```

Comment:
As an aside, I think this is unlikely to ever be an explicit choice, _except_ in the specific case of APIs being renamed/moved between modules, where during a deprecation period, the same procedure is exported under both its old and new paths. This wouldn't have any practical effect on debugging and such, since in both cases the procedure hasn't changed.

It'll be interesting to see how frequently this actually occurs in practice. I do think it would be a good idea for us to have a pedantic warning mode, or some other tooling that would let us analyze things like this on an opt-in basis, but I suspect that in general we're unlikely to see scenarios in which this causes confusion except for in large programs.

Just to be clear, I think this is a good note to have, but I would maybe suggest illustrating the circumstances in which this can occur, and the expected worst-case scenarios. Otherwise it sounds a bit dire, but also isn't clear to the reader whether or not this is something they need to be actively worried about or not, or how to even vet/prevent this from happening.

Reply:
I added to the comment, and created #1479